### PR TITLE
refactor: improve and fix some typings

### PR DESCRIPTION
**Pull request by EdJoPaTo**
I refactored some methods I came by mainly while fixing the typescript compilation. (my other Pull Requests might depend on this PR)

### CODE REVIEW

File: src/utils/build_url.ts [hunk 1]
```diff
@@ -2,20 +2,18 @@ import type { ApiQueryParameters, Url } from '../types/options.js'
...
 
-    // Remove null or undefined parameters
-    Object.keys(queryObj).forEach(key => {
-      if (queryObj[key] == null) delete queryObj[key]
-    })
-
-    return instanceApiEndpoint + '?' + stringifyQuery(queryObj)
+    const queryArray = Object.entries(queryObj)
+      // Remove null or undefined parameters
+      .filter(([ _key, value ]) => typeof value !== 'undefined' && value !== null)
```
Comment (maxlath) [inline src/utils/build_url.ts:13]: I generally prefer the more succinct and strictly(?) equivalent `value != null` for this kind of check
```suggestion
      .filter(([ _key, value ]) => value != null)
```

File: src/utils/build_url.ts [hunk 2]
```diff
@@ -2,20 +2,18 @@ import type { ApiQueryParameters, Url } from '../types/options.js'
...
     // See https://www.wikidata.org/w/api.php ('origin' parameter)
     if (isBrowser) queryObj.origin = '*'
 
-    // Remove null or undefined parameters
-    Object.keys(queryObj).forEach(key => {
-      if (queryObj[key] == null) delete queryObj[key]
-    })
-
-    return instanceApiEndpoint + '?' + stringifyQuery(queryObj)
+    const queryArray = Object.entries(queryObj)
```
Comment (maxlath) [inline src/utils/build_url.ts:11]:
```suggestion
    const queryEntries = Object.entries(queryObj)
```

File: src/utils/build_url.ts [hunk 3]
```diff
@@ -2,20 +2,18 @@ import type { ApiQueryParameters, Url } from '../types/options.js'
...
+      // Remove null or undefined parameters
+      .filter(([ _key, value ]) => typeof value !== 'undefined' && value !== null)
+      .map(([ key, value ]) => [ key, String(value) ])
```
Comment (maxlath) [inline src/utils/build_url.ts:14]: The stringification will already be done by `URLSearchParams#toString`, no?
```suggestion
```

Comment (EdJoPaTo) [reply src/utils/build_url.ts:14]: According to the types the constructor accepts `string | string[][] | Record<string, string> | URLSearchParams` which is why I added that line.

File: src/utils/build_url.ts (hunk 1 above)
Comment (EdJoPaTo) [reply src/utils/build_url.ts:13]: In my head `!=` is a bad practice and I also try to prefer `undefined` over `null` wherever I can so I'm not really sure.

Maybe a better check here would be to check for the actual types `string | number`:

```suggestion
      .filter(([ _key, value ]) => typeof value === 'string' || typeof value === 'number')
```

Comment (maxlath) [reply src/utils/build_url.ts:13]: I'm from the [Douglas Crockford school: always use strict equality, with this one exception to check if a value is null or undefined](https://stackoverflow.com/a/16253116/3324977)

File: src/helpers/simplify_sparql_results.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import type { SimplifiedSparqlResults, SparqlResults } from '../types/sparql.js'
 
-interface SimplifySparqlResultsOptions {
-  minimize?: boolean;
+export type SimplifySparqlResultsOptions = {
```
[the remark this answers is not in the document]
Comment (EdJoPaTo) [reply src/helpers/simplify_sparql_results.ts:3]: `type` is a bit more strict as stuff like `extends` isn't possible. `xo` therefore enforces types. Maybe just enable [this eslint rule](https://typescript-eslint.io/rules/consistent-type-definitions/) for either types or interfaces and stick with it.

File: src/utils/build_url.ts (hunk 1 above)
Comment (EdJoPaTo) [reply src/utils/build_url.ts:13]: It's good to critically question some things. I like that!

**Pull request merged**